src/hooks/search/index.ts

```diff
@@ -218,39 +248,47 @@ const useSearch = (
         }
     }, [router?.isReady]); // Missing query dependency, but that's ok because we only need this on first page load.
 
-    const hasFiltersSet = !!allFilters.length;
+    const hasFiltersSet = !!filters.length;
     const filteredData = (() => {
         if (!data) {
             return [];
         } else if (!hasFiltersSet) {
             return data;
         } else {
             return data.filter(item => {
-                return itemInFilters(item, allFilters);
+                return itemInFilters(item, filters);
             });
         }
     })();
 
-    const numberOfResults = filteredData.length;
-    const shownData = filteredData.slice(0, resultsToShow);
-    const fullyLoaded = resultsToShow >= numberOfResults;
+    const clearAll = () => {
+        setSearchString('');
+        setFilters([]);
+        setSortBy('Most Recent');
+        updatePageMeta();
+    };
 
     return {
-        data: shownData,
-        error,
-        isValidating,
-        resultsToShow,
-        setResultsToShow,
-        allFilters,
-        setAllFilters,
-        onSearch: debouncedOnSearch,
-        onFilter,
-        searchString,
-        setSearchString,
-        fullyLoaded,
-        numberOfResults,
-        onSort,
-        sortBy,
+        searchBoxProps: {
+            searchString,
+            onSearch: debouncedOnSearch,
+        },
+        filterProps: {
+            onFilter,
+            filters,
+        },
+        sortBoxProps: {
+            onSort,
+            sortBy,
+        },
+        resultsProps: {
+            results: filteredData,
+            error,
+            isValidating,
+            searchString,
+            filters,
+        },
+        clearAll,
```

Comment:
You need to return `pageNumber` here also, in case there is a page number passed within getServerSideProps.
It is causing the issue where we visit the page http://localhost:3000/developer/languages/javascript/?page=6 and then click "Load More" and it becomes http://localhost:3000/developer/languages/javascript?page=2.